app/lib/admin/backend.dart

```diff
@@ -382,13 +382,97 @@ class AdminBackend {
         'NOTICE: Redis caches referencing the package will expire given time.');
   }
 
-  Future _deleteWithQuery<T>(Query query) async {
+  /// Removes the specific package version from the Datastore and updates other
+  /// related entities. It is safe to call [removePackageVersion] on an already
+  /// removed version, as the call is idempotent.
+  Future<void> removePackageVersion(String packageName, String version) async {
+    final caller = await _requireAdminPermission(AdminPermission.removePackage);
+
+    _logger.info('${caller.userId} (${caller.email}) initiated the delete '
+        'of package $packageName $version');
```

Comment:
This won't work when called from `app/bin/tools/remove_package.dart`

Reply:
Ack. But we want to move that tool to call the API (in a subsequent PR).

Reply:
Okay...